libmamba/include/mamba/core/pool.hpp

```diff
@@ -22,6 +22,64 @@ namespace spdlog
 
 namespace mamba
 {
+    class MQueue
+    {
+    public:
+        MQueue()
+            : p_queue(new Queue)
+        {
+            queue_init(p_queue);
+        }
+
+        ~MQueue()
+        {
+            queue_free(p_queue);
+        }
+
+        void push(Id id)
+        {
+            queue_push(p_queue, id);
+        }
+
+        void push(Id id1, Id id2)
+        {
+            queue_push2(p_queue, id1, id2);
+        }
+
+        Id* begin()
+        {
+            return &p_queue->elements[0];
+        }
+
+        Id* end()
+        {
+            return &p_queue->elements[p_queue->count];
+        }
+
+        const Id* begin() const
+        {
+            return &p_queue->elements[0];
+        }
+
+        const Id* end() const
+        {
+            return &p_queue->elements[p_queue->count];
+        }
+
+        operator Queue*()
```

Comment:
Did you consider making it explicit? Not sure what's best in this specific case, I'm just wondering every time I see a conversion operator.

Reply:
It makes it simple to use the Queue in interactions with `libsolv`. I think with explicit, we'd lose the `implicit` conversion, e.g. when doing something like `queue_pop(my_mqeue, 1)`, right?

Reply:
Indeed. The main issue with conversion operations is unwanted implicit conversions, but I guess this one is fine as nothing is lost in the conversion.